src/components/button.js

```diff
@@ -97,8 +97,13 @@ const Button = ({ children, classes, variant, size, type, disabled, ...rest }) =
     endIcon: `${sizes.iconPadding.endIcon[size]} ${classes.endIcon}`,
   }
 
+  const handleClick = e => {
+    e.stopPropagation()
+    if('onClick' in rest) rest.onClick()
```

Comment:
2 small things:
1. This is not as robust of a check, since even if `rest.onClick` exists, there's no check to see if it's actually callable
2. Even for a single line within the conditionals, prefer to flesh it out using curly braces (and note the space before/after the opening/closing parentheses)
```js
if (...) {
  rest.onClick
}
```